### DISCUSSION TITLE
Question about ListView implementation

### DISCUSSION THREAD

**Discussion by liancheng** [Textualize/textual#1667] [Q&A] [answered]
I'm trying to build a radio input widget using the `ListView` widget as an example and got puzzled by the following part of the [`ListView.Selected` implementation](https://github.com/Textualize/textual/blob/ab49dce39b51f63fbf3f06e4245cffcda4d36fe5/src/textual/widgets/_list_view.py#L161):

```python
class ListView(Vertical, can_focus=True, can_focus_children=False):
    ...

    class Selected(Message, bubble=True):
        ...
        def __init__(self, sender: ListView, item: ListItem) -> None:   # <-- NOTE
            super().__init__(sender)
            self.item: ListItem = item
```

Please note that `Selected.__init__` referenced `ListView` as part of the argument list. While adding a custom event for my radio input widget in a similar way, I got a "name 'RadioInput' not defined" error. The following simplified snippet reproduces this issue:

```
>>> class A:
...   class B:
...     def __init__(self, a: A):
...       pass
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 2, in A
  File "<stdin>", line 3, in B
NameError: name 'A' is not defined
```

I'm not a Python expert but I guess this makes some sense since the definition of the outer class `A` is incomplete when `A` is referenced in the argument list of `B.__init__`? However, why did it work in the `ListView` case?

(BTW, I mitigated this issue by changing the type of `sender` to `MessageTarget` instead of the concrete widget class.)

**Comment by willmcgugan** [comment, marked as answer]
Try this at the top of your file:

```python
from future import __annotations__
```

**Reply by liancheng** [reply]
Thank you, it worked! Although I guess you meant

```python
from __future__ import annotations
```

Wondered why for a while until I remembered that Python type hints are annotations as well. For future reference, the "Note" in this section explains why https://docs.python.org/3.10/library/typing.html#typing.TYPE_CHECKING
